packages/react-query/src/server/ssgProxy.ts

```diff
@@ -173,12 +205,36 @@ export function createServerSideHelpers<TRouter extends AnyRouter>(
       );
 
       const helperMap: Record<keyof AnyDecoratedProcedure, () => unknown> = {
-        fetch: () => queryClient.fetchQuery({ queryKey, queryFn }),
-        fetchInfinite: () =>
-          queryClient.fetchInfiniteQuery({ queryKey, queryFn }),
-        prefetch: () => queryClient.prefetchQuery({ queryKey, queryFn }),
-        prefetchInfinite: () =>
-          queryClient.prefetchInfiniteQuery({ queryKey, queryFn }),
+        fetch: () => {
+          const args1 = args[1] as Maybe<TRPCFetchQueryOptions<any, any>>;
+          return queryClient.fetchQuery({ ...args1, queryKey, queryFn });
+        },
+        fetchInfinite: () => {
+          const args1 = args[1] as Maybe<
+            TRPCFetchInfiniteQueryOptions<any, any, any>
+          >;
+          return queryClient.fetchInfiniteQuery({
+            ...args1,
+            queryKey,
+            queryFn,
+            defaultPageParam: args1?.initialCursor ?? null,
+          });
+        },
+        prefetch: () => {
+          const args1 = args[1] as Maybe<TRPCFetchQueryOptions<any, any>>;
+          return queryClient.prefetchQuery({ ...args1, queryKey, queryFn });
+        },
+        prefetchInfinite: () => {
+          const args1 = args[1] as Maybe<
+            TRPCFetchInfiniteQueryOptions<any, any, any>
+          >;
+          return queryClient.prefetchInfiniteQuery({
+            ...args1,
+            queryKey,
+            queryFn,
+            defaultPageParam: args1?.initialCursor ?? null,
+          });
```

Comment:
can't really see this in here, but does `fetchInfinite` and `prefetchInfinite` accept the new `pages` params?

https://tanstack.com/query/v5/docs/react/guides/prefetching#prefetching-infinite-queries

Reply:
It's passed in based on the input: https://github.com/trpc/trpc/blob/5f1aee28839fe9fbd19ee240d8af8beb706d2384/packages/react-query/src/internals/context.tsx#L40